src/contents.jl

```diff
@@ -0,0 +1,180 @@
+# end
+
+function contents(auth::Authorization, owner::String, repo::String;
+                    path::String = "", headers = Dict(), ref = nothing,
+                    options...)
+    authenticate_headers(headers, auth)
+    query = Dict()
+    ref == nothing || (query["ref"] = ref)
+    r = get(URI(API_ENDPOINT; path = "/repos/$owner/$repo/contents/$path");
+                    headers = headers, query = query, options...)
+    handle_error(r)
+    return File(JSON.parse(r.data))
+end
+
+function create_file(auth::Authorization, owner::String, repo::String,
+                    path::String, message::String, content; headers = Dict(),
+                    branch::String = nothing, author = nothing,
+                    committer = nothing, options...)
+      data = {"message"=> message, "content"=> content,
+              "author"=> author, "committer"=> committer, "branch" => branch}
+  return upload_file(auth, owner, repo, path, data, headers)
+end
+
+
+function update_file(auth::Authorization, owner::String, repo::String,
+                    path::String, message::String, sha::String, content;
+                    author::User = User({"name"=> "NA", "email"=>"NA"}),
+                    committer::User = User({"name"=> "NA", "email"=>"NA"}),
+                    headers = Dict(), branch = nothing)
+    data = {"message"=> message, "content"=> content,
+            "author"=> author, "committer"=> committer,
+            "sha"=> sha, "branch" => branch}
+  return upload_file(auth, owner, repo, path, data, headers)
+end
+
+function upload_file(auth::Authorization, owner::String, repo::String,
+                    path::String, data, headers)
+    for (k,v) in data
+      v == nothing && delete!(data, k)
+    end
+    data["content"] = bytestring(encode(Base64, JSON.json(data["content"])))
+    authenticate_headers(headers, auth)
+    r = put(URI(API_ENDPOINT; path = "/repos/$owner/$repo/contents/$path"),
+            json=data; headers = headers)
+    handle_error(r)
+    resp = JSON.parse(r.data)
+    return {"content" => File(get(resp, "content", nothing)),
+            "commit" => Commit(get(resp, "commit", nothing))}
+end
+
+
+function delete_file(auth::Authorization, owner::String, repo::String,
+                path::String, sha::String, message::String; branch = "default",
+                headers = Dict(), author::User = User({"name"=> "NA", "email"=>"NA"}),
+                committer::User = User({"name"=> "NA", "email"=>"NA"}))
+    data = {"message" => message, "sha" => sha}
+    branch == "default" || (data["branch"] = branch)
+    author.name == "NA" || (data["author"] = author)
+    committer.name == "NA" || (data["committer"] = committer)
+    authenticate_headers(headers, auth)
+    r = Requests.delete(URI(API_ENDPOINT;
+            path = "/repos/$owner/$repo/contents/$path"),
+            json=data, headers = headers)
+    handle_error(r)
+    # print(get(JSON.parse(r.data), "commit", nothing))
```

Comment:
:) not a big deal, obviously, but you might want to remove this debugging line as well